nvm.sh

```diff
@@ -1907,6 +1908,57 @@ nvm_download_artifact() {
   nvm_echo "${TARBALL}"
 }
 
+# args: tarball path, NVM_OS var
+nvm_extract_tar_artifact() {
+  local tar_compression_flag
+  local TARBALL
+  local TAR_COMMAND_SUFFIX
+  local TMPDIR
+  local NVM_OS
+  local tar
+
+  TARBALL="${1}"
+  NVM_OS="${2}"
+
+  tar='tar'
+  if [ "${NVM_OS}" = 'aix' ]; then
+    tar='gtar'
+  fi
+
+  tar_compression_flag='z'
+
+  if nvm_supports_xz "${VERSION}" && [ "_$NVM_OS" != "_openbsd" ]; then
+    tar_compression_flag='J'
+  fi
+
+  if [ "_$NVM_OS" = "_openbsd" ] && $(which xz >/dev/null 2>&1); then
```

Comment:
this should be using `nvm_has`, and also should be behind `nvm_supports_xz`, otherwise it'll use `xz` on node versions that don't supply xz compression.

```suggestion
  if [ "_$NVM_OS" = "_openbsd" ] && nvm_has xz; then
```